homeassistant/components/roomba/combo.py

```diff
@@ -0,0 +1,159 @@
+"""Class for Combo devices."""
+import logging
+
+from homeassistant.components.vacuum import VacuumEntityFeature
+
+from .irobot_base import SUPPORT_IROBOT, IRobotVacuum
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_BIN_DETECTED = "detected_bin"
+ATTR_PAD_WETNESS = "spray_amount"
+ATTR_NAV_MODE = "navigation_mode"
+ATTR_VACC_POWER = "vacuuming_power"
+
+LOW_POWER = "Low"
+HIGH_POWER = "High"
+
+NAV_REACTIVE = "Reactive"
+NAV_STRAIGHT_LINE = "Straight"
+
+COMBO_LIQUID_ECO = "Eco"
+COMBO_LIQUID_STANDARD = "Standard"
+COMBO_LIQUID_ULTRA = "Ultra"
+
+# NAV_REACTIVE = 0; NAV_STRAIGHT_LIVE = 1
+COMBO_NAVIGATION_MODES = [NAV_REACTIVE, NAV_STRAIGHT_LINE]
+COMBO_VACUUM_POWER = [LOW_POWER, HIGH_POWER]
+COMBO_LIQUID_AMOUNT = [COMBO_LIQUID_ECO, COMBO_LIQUID_STANDARD, COMBO_LIQUID_ULTRA]
+
+SUPPORT_COMBO = SUPPORT_IROBOT | VacuumEntityFeature.FAN_SPEED
+
+
+class RoombaCombo(IRobotVacuum):
+    """Combo."""
+
+    def __init__(self, roomba, blid):
+        """Initialize the Roomba handler."""
+        super().__init__(roomba, blid)
+
+        # Initialize fan speed list
+        speed_list = []
+        for behavior in COMBO_VACUUM_POWER:
+            for spray in COMBO_LIQUID_AMOUNT:
+                for mode in COMBO_NAVIGATION_MODES:
+                    speed_list.append(f"{behavior} - {spray} - {mode}")
+        self._speed_list = speed_list
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_COMBO
+
+    @property
+    def fan_speed(self):
+        """Return the fan speed of the vacuum cleaner."""
+        # Mopping behavior and spray amount as fan speed
+        vac_high = self.vacuum_state.get("vacHigh")
+        behavior = HIGH_POWER
+        if vac_high is False:
+            behavior = LOW_POWER
+        return f"Vacuum mode: {behavior} Power"
+
+    @property
+    def get_fan_speed(self):
+        """Return the actual fan speed."""
+        vac_high = self.vacuum_state.get("vacHigh")
+        behavior = HIGH_POWER
+        if vac_high is False:
+            behavior = LOW_POWER
```

Comment:
Make a helper which returns `HIGH_POWER` or `LOW_POWER` and call it from `fan_speed` and `get_fan_speed`